### PR TITLE
Fix # 1843 : Profile Type with WP Roles are not synched when changing Profile Type

### LINKED ISSUE #1843: Profile Type with WP Roles are not synched when changing Profile Type
**Describe the bug**
Creating Profile Types has the ability to select a WP Role. But when you update Profile Types with different WP Roles, this is not synched or updated

**To Reproduce**
1. Create Profile Types with different WP Roles
 ex. name: Profile Type 1, role: contributor
      name: Profile Type 2, role: author
2. Create a sample user
3. Assign the user to Profile Type 1, in WP Dashboard > Users
4. Assign the user to Profile Type 2
5. See error, when assigned to Profile Type 2, user should be author

**Expected behavior**
When Profile Tyep1 with Contributor role is selected for a user and changed to Profile Type 2.
The WP Role should also changed as Profile Type 2 has Author role.

**Screenshots**
https://drive.google.com/file/d/1SlQWoKYwyosipaZdiMPGAmCc-mPabUXm/view?usp=sharing
issue can be seen in demo
https://drive.google.com/file/d/14P1xArYPwBICZRdmmi9XC_9RXlM-1i7x/view

**Support ticket links**
If applicable, add HelpScout link or ticket number where the issue was originally reported.

### PR DESCRIPTION
**Pull request by manishpatolia**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #1843.

### How to test the changes in this Pull Request:

1. Create Profile Types with different WP Roles
 ex. name: Profile Type 1, role: contributor
      name: Profile Type 2, role: author
2. Create a sample user
3. Assign the user to Profile Type 1, in WP Dashboard > Users list
4. Assign the user to Profile Type 2

### Proof Screenshots or Video

https://www.loom.com/share/b9f533fd58464e418c58472aa0df963c

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

File: src/bp-members/classes/class-bp-members-admin.php [hunk 1]
```diff
@@ -2674,6 +2674,72 @@ public function users_table_process_bulk_type_change() {
...
+						 * For add the new profile type roles
+						 **/
+						if ( !$error ) {
+							// Get post id of selected profile type.
+							$post_id = bp_member_type_post_by_type( $new_type );
+							
+							// Get selected profile type role.
+							$selected_member_type_wp_roles = get_post_meta( $post_id, '_bp_member_type_wp_roles', true );
+							$member_type_role    = sanitize_text_field( $selected_member_type_wp_roles[0] );
+							$member_type_role    = ( isset( $member_type_role ) ) ? $member_type_role : '';
```
Comment (chetansatasiya) [inline src/bp-members/classes/class-bp-members-admin.php:2688]: @manishpatolia follow the proper indentaion of code

**Review by chetansatasiya — CHANGES_REQUESTED** [review]
Follow the code indentaion

File: src/bp-members/classes/class-bp-members-admin.php (hunk 1 above)
Comment (manishpatolia) [reply src/bp-members/classes/class-bp-members-admin.php]: Changed

**Review by chetansatasiya — APPROVED** [review]

**Review by KartikSuthar — APPROVED** [review]

**Comment by JetalDobariya** [comment]
Hello @manishpatolia 

I tested this PR in fix branch fix/1843 and it is working fine.
See Proofs : https://www.loom.com/share/e828db7387fc4d8abae749f168cab8a1

Please merge this code in qa-ongoing branch.

Thanks!

**Comment by chetansatasiya** [comment]
> Hello @manishpatolia
> 
> I tested this PR in fix branch fix/1843 and it is working fine. See Proofs : https://www.loom.com/share/e828db7387fc4d8abae749f168cab8a1
> 
> Please merge this code in qa-ongoing branch.
> 
> Thanks!

@JetalDobariya merged in qa-ongoing

**Comment by JetalDobariya** [comment]
Hello @chetansatasiya 

I did verify this PR in qa-ongoing branch, and it is working fine.
image

CC : @manishpatolia 
Thanks!

**Comment by JetalDobariya** [comment]
Hello @manishpatolia 

I did verify this PR in qa-approved branch, Roles has been updated with Profile type.
Befor change Profile type Role : https://prnt.sc/1zrbcbj
After changed Profile type : https://prnt.sc/1zrbtkl

Thanks!
CC : @sourabhmatolia @firozwp

**Pull request merged**